Update to mongo 4.0

Ameausoone opened the pull request:
Update MongoDB version to 3.4
Use docker volumes instead of local storage
Use default standard MongoDB ports
Use wait-for-it.sh to wait until Mongo servers are available (but there is better way to wait until servers are fully ready)
Add Makefile 
Update documentation

Ameausoone committed:
Upgrade mongo to version 3.4
Use new config servers replica set
Use docker volumes
Use default port

Ameausoone committed:
Update documentation

rlespinasse committed:
feat(makefile): auto generated help

- generate automatically the help 
- make help, the default task

Ameausoone committed:
Merge pull request #1 from rlespinasse/auto-generated-help

feat(makefile): auto generated help

Ameausoone committed:
Update README.md

andiempett commented:
when i try this pull request out, i find that mongos1_1 fails to start up with the error
```
...
...failed Permission denied for socket: /tmp/mongodb-27017.sock
...Failed to set up sockets during startup.
...shutting down with code:48
```

RemyG committed:
fix(doc): Fix typo in README

Ameausoone committed:
fix(doc): Fix typo in README

RemyG committed:
fix(docker): Fix startup error

Fix error ```failed Permission denied for socket: /tmp/mongodb-27017.sock``` when starting on Linux.

Ameausoone committed:
feat(version): update to mongo 4.0

remove custom volumes management and let docker manage it
some fixes to update to mongo 4.0 (bind_ip_all, nojournal, rest)
update documentation

Ameausoone committed:
Merge branch 'master' into patch-2

Ameausoone committed:
fix(docker): Fix startup error
